check_logging.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+DIFF=$(git diff origin/main...HEAD || true)
```

Comment:
This diff should take into account when the pull request is against other branches that are not `main`, otherwise it will be making comparisons not only the pull request changes but also any diff between the current branch and `main`

Reply:
Most of the PRs for WildFly are against `main` branch only, in that case, if we stop the diff check then it'll pass automatically. And also, I'm not sure how we can fetch the details of a pull request to check which base branch it's targeting. Can you please explain a bit?

Reply:
> Most of the PRs for WildFly are against main branch only, in that case, if we stop the diff check then it'll pass automatically.

We produce micro releases, so this should work also with these branches without any manual intervention.


>  I'm not sure how we can fetch the details of a pull request to check which base branch it's targeting.

I have no so much idea either, searching for information about that it looks like that `github.base_ref` property can provide you this information:

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs

github.base_ref: The base_ref or target branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is either pull_request or pull_request_target.

This was a quick search only, I'm not sure even if that is exactly what we need, but looks promising. So, it is a matter of investigating it.

Reply:
@yersan Thanks! I have fixed it now according to our conversation. Additionally, I have also added the below code in GitHub Action so that we can see base branch is correctly fetched or not. I'll remove the part once you think it's correctly done.

```
- name: Print GitHub context
    run: echo "${{ toJson(github) }}"
```
output:        

    "head_ref": "WFLY-19605",
    "base_ref": "main",

Reply:
Thanks @RanabirChakraborty , looking at the code, this looks fixed. 

Is there any proof of how this change works in action? I mean any test done by you where invalid logs are found and so one? That for sure will help to the reviewers.

Notice other revision comments have not been resolved yet.

Reply:
+1 as suggested [above](https://github.com/wildfly/wildfly/pull/18116#discussion_r1722055354), would be nice to see this in action in a private fork example PR.

Reply:
@yersan @rhusar, according to your suggestion to test this GitHub Action, I created a dummy repo(https://github.com/RanabirChakraborty/spring-demo) with this GitHub Action present and sent a PR with the non-i18n INFO/WARN/ERROR logging and you can see the action is failing for the PR - https://github.com/RanabirChakraborty/spring-demo/pull/1, You can also check it by yourself.

Reply:
Thanks, @RanabirChakraborty, look good from my side, I'm done with my review